### PR TITLE
rudimentary end to end demo using conjur-oss

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
#### What ticket does this PR close?
Connected to #790 
#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run the [K8s demo](https://github.com/cyberark/secretless-broker/tree/master/demos/k8s-demo) (which requires changing the image the demo uses to the local build of Secretless)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [ ] An updated README with instructions on how to manually test this feature
- [ ] Utility `start` and `stop` scripts to spin up and tear down the test environments
- [ ] A `test` script to run some basic manual tests (optional; if does not exist, the README should have detailed instructions)

### CODE REVIEW

File: demos/conjur-oss/requirements/conjur/03-apply-app-policy.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/conjur/03-apply-app-policy.sh]: Needs `set -e`

File: demos/conjur-oss/requirements/conjur/app-policy.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/conjur/app-policy.sh]: Needs `set -e` and undefined var flag

File: demos/conjur-oss/requirements/conjur/conjur-policy.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/conjur/conjur-policy.sh]: Needs `set -e` and undefined var flag

File: demos/conjur-oss/requirements/conjur/env.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# configurable env vars
+AUTHENTICATOR_ID=example
+
+APP_SECRETS_POLICY_BRANCH=apps/secrets/test
+APP_SECRETS_READER_LAYER=apps/layers/myapp
+
+CONJUR_ACCOUNT=example_acc
+
+OSS_CONJUR_SERVICE_ACCOUNT_NAME=conjur-sa
+OSS_CONJUR_NAMESPACE=kumbi-conjur
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/conjur/env.sh]:
```suggestion
OSS_CONJUR_NAMESPACE=conjur-instance
```

Comment (sgnn7) [reply demos/conjur-oss/requirements/conjur/env.sh]: Any reason why we keep redeclaring these same vars?

File: demos/conjur-oss/requirements/gen-env.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/gen-env.sh]: Needs `set -e` and undefined var flag

File: demos/conjur-oss/requirements/mysql/-00-clean.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/mysql/-00-clean.sh]: As mentioned in a previous comment, this should be `./stop`

File: demos/conjur-oss/requirements/mysql/00-deploy-mysql.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/mysql/00-deploy-mysql.sh]: Needs `set -e` and undefined var flag

File: demos/conjur-oss/requirements/mysql/-00-clean.sh (hunk 6 above)
Comment (sgnn7) [inline demos/conjur-oss/requirements/mysql/-00-clean.sh]: Needs undefined var flag

File: demos/conjur-oss/requirements/mysql/env.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+# configurable env vars
+MYSQL_NAMESPACE="kt-mysql"
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/mysql/env.sh]:
```suggestion
MYSQL_NAMESPACE="mysql-instance"
```

File: demos/conjur-oss/requirements/mysql/env.sh [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+MYSQL_RELEASE="kt-mysql-release"
```
Comment (sgnn7) [inline demos/conjur-oss/requirements/mysql/env.sh]:
```suggestion
MYSQL_RELEASE="mysql-release"
```

File: demos/conjur-oss/00-prepare-application.sh [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+# ensure conjur authenticator is able to operate in application namespace
+# via kubernetes RBAC
+./conjur-authenticator-role.sh > tmp/conjur-authenticator-role.yml
+kubectl create -f tmp/conjur-authenticator-role.yml
+
+# create application namespace
+kubectl create ns "${APP_NAMESPACE}"
+
+# create CLI container
+kubectl --namespace "${APP_NAMESPACE}" run conjur-cli --image=cyberark/conjur-cli:5 --restart=Never --attach=false --command -- sleep infinity
```
[the remark this answers is not in the document]
Comment (doodlesbykumbi) [reply demos/conjur-oss/00-prepare-application.sh]: `--rm=true` is only for attached containers

File: demos/conjur-oss/00-prepare-application.sh [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+cat << EOL | kubectl --namespace "${APP_NAMESPACE}" exec -i conjur-cli -- bash -
```
[the remark this answers is not in the document]
Comment (doodlesbykumbi) [reply demos/conjur-oss/00-prepare-application.sh]: heredocs have the benefit that i don't have to worry about escaping stuff (single and double quotes etc.)

File: demos/conjur-oss/00-prepare-application.sh [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+# API key here is the key that creation of the account provided you in step #2
+conjur authn login -u '${CONJUR_ADMIN_AUTHN_LOGIN}' -p '${CONJUR_ADMIN_API_KEY}'
+
+# Check that you are identified as the admin user
+conjur authn whoami
+
+EOL
+
+# retrieve conjur SSL certificate
+kubectl --namespace "${APP_NAMESPACE}" exec -i conjur-cli -- bash -c "cat /root/*.pem" > tmp/conjur.pem
```
[the remark this answers is not in the document]
Comment (doodlesbykumbi) [reply demos/conjur-oss/00-prepare-application.sh]: i can but i don't know the name of the file beforehand thus the `*`... not sure how copy works with that

**Pull request closed without merging**